FreeWillPBC/flywaydb-cli
# OPS-2175: migrating usage of request and request-promise to axios

[OPS-2175](https://freewill.atlassian.net/browse/OPS-2175)

## How did you implement it
moving all usage of `request` and `request-promise` (and subsequently `request-progress`) to use `axios` instead. 

`request` and `request-promise` are prone to prototype pollution vulnerability from versions of `tough-cookie` older than 4.1.3 (more info in ticket)

[OPS-2175]: https://freewill.atlassian.net/browse/OPS-2175?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## How can we verify it
`getReleaseSource` and `downloadFlywaySource` are called when we run `npm run install` in the flyway project (after building with our new changes of course). `getReleaseSource` should return the correct data from the Maven repository and is the first step in `npm run install` command. `downloadFlywaySource` is the second step in `npm run install` command and will save a file to your machine after success. Here are ss's of running `npm run install` locally (same output as when we used request library, and also got verbose error logging as expected when I had errors in my code!):

in the middle, we can see the progress bar working as intended:
![image](https://github.com/user-attachments/assets/88220b52-edbc-4408-8057-43ba0b2d2137)

when completed:
![image](https://github.com/user-attachments/assets/28ba04da-fdd8-46b8-8f7a-35abd7474c68)

- test the behavior across different platforms (win32, linux, darwin, and arm64) to confirm no regression
  - how might I do this?
